integration/test_collection_filter.py

```diff
@@ -411,3 +413,104 @@ def test_ref_filters_multi_target(client: weaviate.Client):
+            DataObject(properties={"description": "Loads of money", "age": 60}, uuid=uuid.uuid4()),
+            DataObject(properties={"description": "Lots of money", "age": 40}, uuid=uuid.uuid4()),
+        ]
+    )
+    objs = collection.data.get()
+    assert len(objs) == 6
+
+    collection.data.delete_many(where=Filter(path="name").equal("some name"), verbose=True)
+    objs = collection.data.get()
+    assert len(objs) == 5
+
+    collection.data.delete_many(where=Filter(path="name").equal("some other name"))
+    objs = collection.data.get()
+    assert len(objs) == 4
+
+    collection.data.delete_many(where=Filter(path="age").equal(10))
+    objs = collection.data.get()
+    assert len(objs) == 3
+
+    collection.data.delete_many(where=Filter(path="age").is_none(True))
+    objs = collection.data.get()
+    assert len(objs) == 2
+
+    collection.data.delete_many(where=Filter(path="description").equal("money"))
+    objs = collection.data.get()
+    assert len(objs) == 0
+
+
+def test_delete_many_and(client: weaviate.Client):
+    name = "TestDeleteManyAnd"
+    collection = client.collection.create(
+        name=name,
+        properties=[
+            Property(name="Name", data_type=DataType.TEXT),
+            Property(name="Age", data_type=DataType.INT),
+        ],
+        vectorizer_config=VectorizerFactory.none(),
+    )
+    collection.data.insert_many(
+        [
+            DataObject(properties={"age": 10, "name": "Timmy"}, uuid=uuid.uuid4()),
+            DataObject(properties={"age": 10, "name": "Tommy"}, uuid=uuid.uuid4()),
+        ]
+    )
+    objs = collection.data.get()
+    assert len(objs) == 2
+
+    collection.data.delete_many(
+        where=Filter(path="age").equal(10) & Filter(path="name").equal("Timmy")
+    )
+    objs = collection.data.get()
+    assert len(objs) == 1
+    assert objs[0].properties["age"] == 10
+    assert objs[0].properties["name"] == "Tommy"
+
+
+def test_delete_many_or(client: weaviate.Client):
+    name = "TestDeleteManyOr"
+    collection = client.collection.create(
+        name=name,
+        properties=[
+            Property(name="Name", data_type=DataType.TEXT),
+            Property(name="Age", data_type=DataType.INT),
+        ],
+        vectorizer_config=VectorizerFactory.none(),
+    )
+    collection.data.insert_many(
+        [
+            DataObject(properties={"age": 10, "name": "Timmy"}, uuid=uuid.uuid4()),
+            DataObject(properties={"age": 20, "name": "Tim"}, uuid=uuid.uuid4()),
+            DataObject(properties={"age": 30, "name": "Timothy"}, uuid=uuid.uuid4()),
+        ]
+    )
+    objs = collection.data.get()
+    assert len(objs) == 3
+
+    collection.data.delete_many(where=Filter(path="age").equal(10) | Filter(path="age").equal(30))
+    objs = collection.data.get()
+    assert len(objs) == 1
+    assert objs[0].properties["age"] == 20
+    assert objs[0].properties["name"] == "Tim"
```

Comment:
I think we need tests for each datatype - we need to be sure that the correct text for the REST-filter is picked